templates/add-persistence-attribute-PASSWORD_RESET_POST_CHALLENGE/code.js

```diff
@@ -63,6 +62,51 @@ exports.onExecutePostChallenge = async (event, api) => {
         }
     );
 };
+/**
+ * Get an AccessToken
+ *
+ * @param {CacheAPI} cache
+ * @param {{domain: string, clientId: string, clientSecret: string, audience?: string}} options - AuthenticationClient options to fetch the token
+ * @return {Promise<string>}
+ */
+async function getAccessToken(cache, options) {
+    const key = `access_token_${options.clientId}_${options.audience}`;
+    // Check the cache if we have a valid entry
+    const record = cache.get(key);
+    if (record && record.expires_at > Date.now()) {
+        return record.value;
+    }
+
+    // Get the AccessToken using a client_credential grant.
+    const authClient = new AuthenticationClient(options);
+
+    const {
+        data: { access_token, expires_in },
+    } = await authClient.oauth.clientCredentialsGrant({
+        audience: options.audience ?? `https://${options.domain}/api/v2/`,
+    });
+
+    // Try to cache it
+    const cacheSetResult = cache.set(key, access_token, { ttl: expires_in });
+    if (cacheSetResult.type === 'error') {
+        console.error(`Failed to set ${key}: ${cacheSetResult.code}`);
```

Comment:
@turcottedanny can you confirm that `cacheSetResult.code` is unlikely to log anything "sensitive"?

Reply:
Danny -  These are well known values that we expose to the customer's code, so yes it should be safe